token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,177 @@
+"""
+A module discovers and invoke Token Dispenser Service (TDS) lambda
+"""
+import logging
+import json
+import re
+from typing import List
+from botocore.exceptions import ClientError
+import boto3
+lambda_client = boto3.client('lambda')
+ssm = boto3.client('ssm')
+# ssm parameter store default path for TDS ARN
+_DEFAULT_SSM_PATH_: str = '/service/token-dispenser'
+logger = logging.getLogger(__name__)
+
+
+def get_parameter_by_name(name: str) -> str:
+    """
+
+    Args:
+        name: the ssm name point to TDS arn
+
+    Returns: tds arn as string
+    """
+    try:
+        if not name.endswith('/'):  # Check if it's a specific parameter name
+            response = ssm.get_parameter(
+                Name=name
+            )
+        return response['Parameter']['Value']
+    except KeyError as key_error:
+        logger.exception("Missing Parameter or Value key with name:%s response:%s",
+                         name, response)
+        raise key_error
+    except Exception as ex:  # Catch any other error.
+        logger.exception("An unexpected error occurred during get_parameter: "
+                         "name:%s response:%s", name, response)
+        raise ex
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    if not lambda_arn:
+        return "Error: lambda_arn not provided"
+
+    try:
+        json.loads(input_params_json)
+    except json.JSONDecodeError:
+        return f'Error: Invalid JSON : {input_params_json}'
```

Comment:
We don't need validation here. Assume it's valid since this is an internal function